pandas/core/frame.py

```diff
@@ -8265,26 +8265,22 @@ def to_timestamp(self, freq=None, how="start", axis=0, copy=True) -> "DataFrame"
         -------
         DataFrame with DatetimeIndex
         """
-        new_data = self._data
-        if copy:
-            new_data = new_data.copy()
+        new_obj = self.copy(deep=copy)
 
         axis = self._get_axis_number(axis)
         if axis == 0:
-            assert isinstance(self.index, (ABCDatetimeIndex, ABCPeriodIndex))
-            new_data.set_axis(1, self.index.to_timestamp(freq=freq, how=how))
+            assert isinstance(self.index, ABCPeriodIndex)
+            new_obj.index = self.index.to_timestamp(freq=freq, how=how)
         elif axis == 1:
-            assert isinstance(self.columns, (ABCDatetimeIndex, ABCPeriodIndex))
-            new_data.set_axis(0, self.columns.to_timestamp(freq=freq, how=how))
+            assert isinstance(self.columns, ABCPeriodIndex)
+            new_obj.columns = self.columns.to_timestamp(freq=freq, how=how)
         else:  # pragma: no cover
             raise AssertionError(f"Axis must be 0 or 1. Got {axis}")
 
-        return self._constructor(new_data)
+        return new_obj
 
     def to_period(self, freq=None, axis=0, copy=True) -> "DataFrame":
         """
-        Convert DataFrame from DatetimeIndex to PeriodIndex.
```

Comment:
I think you should undo this removal. (PEP257).

Reply:
ok, though im on record as disliking this sentence fragment